web-bundle/src/main/java/com/graphhopper/http/IllegalArgumentExceptionMapper.java

```diff
@@ -1,14 +1,21 @@
 package com.graphhopper.http;
 
 import com.graphhopper.MultiException;
+import com.graphhopper.util.Helper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import javax.ws.rs.core.Response;
 import javax.ws.rs.ext.ExceptionMapper;
+import javax.ws.rs.ext.Provider;
 
+@Provider
 public class IllegalArgumentExceptionMapper implements ExceptionMapper<IllegalArgumentException> {
+    private static final Logger logger = LoggerFactory.getLogger(IllegalArgumentExceptionMapper.class);
 
     @Override
     public Response toResponse(IllegalArgumentException e) {
+        logger.info("bad request: " + (Helper.isEmpty(e.getMessage()) ? "unknown reason" : e.getMessage()));
```

Comment:
Do you know a way to attach the request path and query parameters?

Reply:
Apparently the full request url is included in the thread name, so when the logging pattern includes `%thread` the url will be included in the logs. For requestLog there is `%r`.